integration/storagetest/logtests.go

```diff
@@ -453,3 +451,85 @@ func createFakeLeaf(ctx context.Context, s storage.LogStorage, tree *trillian.Tr
 
 	return q[0].Leaf
 }
+
+func (*logTests) TestDequeueLeaves(ctx context.Context, t *testing.T, s storage.LogStorage, as storage.AdminStorage) {
+	const leavesToInsert = 5
+	tree := mustCreateTree(ctx, t, as, storageto.LogTree)
+	mustSignAndStoreLogRoot(ctx, t, s, tree, 0)
+
+	leaves := createTestLeaves(leavesToInsert, 20)
+	if _, err := s.QueueLeaves(ctx, tree, leaves, fakeDequeueCutoffTime); err != nil {
+		t.Fatalf("Failed to queue leaves: %v", err)
+	}
+
+	// Now try to dequeue them
+	// Some dequeue implementations probabalistically dequeue and require retrying until timeout.
+	leaves2 := dequeueLeaves(ctx, t, s, tree, fakeDequeueCutoffTime, 5, 0)
+	if len(leaves2) != leavesToInsert {
+		t.Fatalf("Dequeued %d leaves but expected to get %d", len(leaves2), leavesToInsert)
+	}
+
+	// If we dequeue again then we should now get nothing
+	if err := s.ReadWriteTransaction(ctx, tree,
+		func(ctx context.Context, tx3 storage.LogTreeTX) error {
+			leaves3, err := tx3.DequeueLeaves(ctx, 99, fakeDequeueCutoffTime)
+			if err != nil {
+				t.Fatalf("Failed to dequeue leaves (second time): %v", err)
+			}
+			if len(leaves3) != 0 {
+				t.Fatalf("Dequeued %d leaves but expected to get %d", len(leaves3), leavesToInsert)
+			}
+			return nil
+		}); err != nil {
+		t.Fatalf("Could not dequeue the expected number of leaves: %v", err)
```

Comment:
The indentation is confusing here. I'd store `err` without `if`, and `if err != nil` in the next line.

Reply:
Agree, fixed it.